mlrun/feature_store/feature_set.py

```diff
@@ -391,6 +394,50 @@ def purge_targets(self, target_names: List[str] = None, silent: bool = False):
             self, mlrun.api.schemas.AuthorizationAction.delete
         )
 
+        purge_targets = self._reload_and_get_status_targets(
+            target_names=target_names, silent=silent
+        )
+
+        if purge_targets:
+            # TODO - benb - run_uuid shouldn't be a part of purge - remove logic after toggle implementation
+            from mlrun.datastore.targets import generate_target_run_uuid
+
+            run_uuid = generate_target_run_uuid()
+            purge_target_names = list(purge_targets.keys())
+            for target_name in purge_target_names:
+                target = purge_targets[target_name]
+                driver = get_target_driver(target_spec=target, resource=self)
+                try:
+                    driver.purge()
+                    driver.run_uuid = run_uuid
+                except FileNotFoundError:
+                    pass
```

Comment:
If purge raises a `FileNotFoundError`, is it ok not to set the run ID?

Reply:
re-thinking it there is no need in run_id manipulation here - removed